arbnode/batch_poster.go

```diff
@@ -427,6 +463,39 @@ func AccessList(opts *AccessListOpts) types.AccessList {
 	return l
 }
 
+// Adds a block merkle proof to an Espresso justification, providing a proof that a set of transactions
+// hashes to some light client state root
+func (b *BatchPoster) addEspressoBlockMerkleProof(
+	msg *arbostypes.MessageWithMetadata,
+) (*arbostypes.MessageWithMetadata, error) {
```

Comment:
I think this function doesn't really need to return `msg` because it modifies it.